pkg/apis/vpc/v1alpha1/subnetconnectionbindingmap_types.go

```diff
@@ -0,0 +1,63 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:validation:XValidation:rule="!(has(self.targetSubnetSet) && has(self.targetSubnet))",message="Only one of targetSubnetSet or targetSubnet can be specified"
+type SubnetConnectionBindingMapSpec struct {
+	// SubnetName is Subnet name which this binding is attached on.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="subnetName is immutable"
+	SubnetName string `json:"subnetName"`
+	// TargetSubnetSet specifies the target subnetSet which a subnet is bound to.
+	// +kubebuilder:validation:Optional
+	TargetSubnetSet string `json:"targetSubnetSet,omitempty"`
+	// TargetSubnet specifies the target subnet which a subnet is bound to.
+	// +kubebuilder:validation:Optional
+	TargetSubnet string `json:"targetSubnet,omitempty"`
+	// VLANTrafficTag is the vlan tag configured in the binding. It can be empty, then the handler will choose a valid
+	// value based on the existing subnet connection binding maps bound on the target subnet or subnetSet.
```

Comment:
You meant NSX OP will auto allocate a VLAN ID? I suggest not doing that, at least not for this release.